src/Trace/TraceClient.php

```diff
@@ -0,0 +1,199 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Trace;
+
+use Google\Cloud\Core\ArrayTrait;
+use Google\Cloud\Core\ClientTrait;
+use Google\Cloud\Core\Exception\NotFoundException;
+use Google\Cloud\Core\Exception\ServiceException;
+use Google\Cloud\Trace\Connection\ConnectionInterface;
+use Google\Cloud\Trace\Connection\Rest;
+
+/**
+ * Google Stackdriver Trace client. Allows you to collect latency data from
+ * your applications and display it in the Google Cloud Platform Console.
+ * Find more information at [Stackdriver Trace API docs](https://cloud.google.com/trace/docs/).
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Trace\TraceClient;
+ *
+ * $trace = new TraceClient();
+ * ```
+ */
+class TraceClient
+{
+    use ArrayTrait;
+    use ClientTrait;
+
+    const FULL_CONTROL_SCOPE = 'https://www.googleapis.com/auth/cloud-platform';
+    const READ_ONLY_SCOPE = 'https://www.googleapis.com/auth/trace.readonly';
+
+    /**
+     * @var ConnectionInterface $connection Represents a connection to Trace
+     */
+    protected $connection;
+
+    /**
+     * Create a Trace client.
+     *
+     * @param array $config [optional] {
+     *     Configuration options.
+     *
+     *     @type string $projectId The project ID from the Google Developer's
+     *           Console.
+     *     @type CacheItemPoolInterface $authCache A cache used storing access
+     *           tokens. **Defaults to** a simple in memory implementation.
+     *     @type array $authCacheOptions Cache configuration options.
+     *     @type callable $authHttpHandler A handler used to deliver Psr7
+     *           requests specifically for authentication.
+     *     @type callable $httpHandler A handler used to deliver Psr7 requests.
+     *           Only valid for requests sent over REST.
+     *     @type array $keyFile The contents of the service account credentials
+     *           .json file retrieved from the Google Developer's Console.
+     *           Ex: `json_decode(file_get_contents($path), true)`.
+     *     @type string $keyFilePath The full path to your service account
+     *           credentials .json file retrieved from the Google Developers
+     *           Console.
+     *     @type int $retries Number of retries for a failed request.
+     *           **Defaults to** `3`.
+     *     @type array $scopes Scopes to be used for the request.
+     * }
+     */
+    public function __construct(array $config = [])
+    {
+        if (!isset($config['scopes'])) {
+            $config['scopes'] = [self::FULL_CONTROL_SCOPE];
+        }
+
+        $this->connection = new Rest($this->configureAuthentication($config));
+    }
+
+    /**
+     * Returns the projectId for this client.
+     *
+     * @return string
+     */
+    public function projectId()
```

Comment:
You can get the project ID `ClientTrait::configureAuthentication`.

```php
class TraceClient
{
    public function __construct(array $config = []) {
         $config = $this->configureAuthentication($config);
         $this->projectId = $config['projectId'];
    }
}
````

From there, you could manage it internally and hide it from the user.